src/realm/error_codes.h

```diff
@@ -183,18 +183,12 @@ typedef enum realm_errno {
     RLM_ERR_MAINTENANCE_IN_PROGRESS = 4352,
     RLM_ERR_USERPASS_TOKEN_INVALID = 4353,
 
-    RLM_ERR_WEBSOCKET_GOINGAWAY = 4400,
-    RLM_ERR_WEBSOCKET_PROTOCOLERROR = 4401,
-    RLM_ERR_WEBSOCKET_UNSUPPORTEDDATA = 4402,
-    RLM_ERR_WEBSOCKET_RESERVED = 4403,
-    RLM_ERR_WEBSOCKET_NOSTATUSRECEIVED = 4404,
-    RLM_ERR_WEBSOCKET_ABNORMALCLOSURE = 4405,
-    RLM_ERR_WEBSOCKET_INVALIDPAYLOADDATA = 4406,
-    RLM_ERR_WEBSOCKET_POLICYVIOLATION = 4407,
-    RLM_ERR_WEBSOCKET_MESSAGETOOBIG = 4408,
-    RLM_ERR_WEBSOCKET_INAVALIDEXTENSION = 4409,
-    RLM_ERR_WEBSOCKET_INTERNALSERVERERROR = 4410,
-    RLM_ERR_WEBSOCKET_TLSHANDSHAKEFAILED = 4411,
+    RLM_ERR_WEBSOCKET_RESOLVE_FAILED = 4400,
```

Comment:
We could consider adding a new enum just for websocket related error/close codes and remove the one in error_codes.hpp. Wouldn't that work?

Reply:
Also, merge the ones here with WebSocketError since they all fit together

Reply:
I guess the main problem is that the codes will have new values, since 1xxx may be be used already for example. For server values, we could translate them in websocket.cpp and pass the new values along. Of course, a custom websocket implementation would have to do the same.